client/src/client.ts

```diff
@@ -3190,6 +3221,11 @@ export abstract class BaseLanguageClient {
 		this._diagnostics.set(uri, diagnostics);
 	}
 
+	private getFolderForFileBasedCancellation(folderName?: string) {
+		// client and server must use same logic to create actual folder name. but don't have a good way to share logic.
+		return folderName ? path.join(os.tmpdir(), 'vscode-languageserver-cancellation', folderName) : undefined;
```

Comment:
what would be best way to share it between client and server? only thing shared between 2 are protocol and I dont want to expose it there.

Reply:
See comments above plus I would expose on MessageConnection a property containing the cancellationFolder if file based cancelation was requested. That folder name can the be passes as an argument to the server (on the command line)

Reply:
hmmm... I meant this rule
..
os.tmpdir()/vscode-languageserver-cancellation/{this folder name is passed as command line argument}
..

the last part is already passed as command-line argument.

Reply:
I would like to keep "vscode-languageserver-cancellation' folder since it makes it easy to spot this folder. typescript is using the same strategy and it makes investigation easier.